### DISCUSSION TITLE
Can we customise the router?

### DISCUSSION THREAD

**Discussion by joewearing** [nuxt/nuxt#18273] [Questions]
I'm not taking about `@nuxtjs/router`

I was hoping there was a way to configure the inbuilt one, like we could in Nuxt 2 [nuxt 2 docs](https://nuxtjs.org/docs/configuration-glossary/configuration-router/)

I would like to add a `beforeEach` function specifically, which I was able to do in Nuxt 2 by creating a Nuxt plugin and accessing the router using `app.router`

**Comment by visini** [comment]
You can follow `@nuxtjs/router` status via https://github.com/nuxt/framework/discussions/751

**Comment by kissu** [comment]
I didn't tried this one yet but you can maybe try [@nuxtjs/router-extras](https://github.com/nuxt-community/router-extras-module) too.
